core/build_structs.go

```diff
@@ -67,6 +67,7 @@ type TargetSpecific struct {
 type dependentInterface interface {
 	phonyInterface
 	outputDir(g generatorBackend) string
+	implicitOutputs(g generatorBackend) []string
```

Comment:
Would this not make more sense in `phonyInterface` next to `outputs()`? It seems a bit strange to have different kinds of outputs in different interfaces.

(Admittedly I think both should be in `dependentInterface`, but we shouldn't change that here).

Reply:
I agree that it looks odd. However, I don't think implicitOutputs belongs in phonyInterface - phony support only requires outputs()

Reply:
Thinking about this again, it's possible to define a module that only has implicit outputs (no explicit outputs). A phony rule must therefore depend on both the explicit and implicit outputs.

So please add implicitOutputs() to the phonyInterface, and update phony target generation to include implicitOutputs(). Eventually we might want an allOutputs() function.